### PR TITLE
Update to use latest versions of everything

### PR DESCRIPTION
**Pull request by samoht**
Especially, use MirageOS 3 signatures instead of mirage-types.2.8.

### CODE REVIEW

File: lib/dns_forward_cache.ml [hunk 1]
```diff
@@ -84,7 +84,7 @@ module Make(Time: V1_LWT.TIME) = struct
     let min_ttl = List.fold_left (min) Int32.max_int (List.map (fun rr -> rr.Dns.Packet.ttl) rrs) in
     let timeout =
       let open Lwt.Infix in
-      Time.sleep (Int32.to_float min_ttl)
+      Time.sleep_ns (Int64.of_int32 min_ttl)
```
Comment (djs55) [inline lib/dns_forward_cache.ml]: Is `min_ttl` in nanoseconds? I think it might be in seconds.

File: lib/dns_forward_error.ml [hunk 2]
```diff
@@ -18,13 +18,13 @@ open Lwt.Infix
...
-module FromFlowError(Flow: V1_LWT.FLOW) = struct
+let errorf fmt = Fmt.kstrf (fun s -> Lwt.return (Error (`Msg s))) fmt
+
+module FromFlowError(Flow: Mirage_flow_lwt.S) = struct
   let (>>=) m f = m >>= function
-    | `Eof -> Lwt.return (Result.Error (`Msg "Unexpected end of file"))
-    | `Error e -> Lwt.return (Result.Error (`Msg (Flow.error_message e)))
-    | `Ok x -> f x
+    | `Eof     -> errorf "Unexpected end of file"
+    | `Error e -> errorf "%a" Flow.pp_error e
```
Comment (djs55) [inline lib/dns_forward_error.ml]: Thanks -- I'm still not used to using `Fmt`!

File: lib/dns_forward_cache.ml (hunk 1 above)
Comment (samoht) [inline lib/dns_forward_cache.ml]: ha yes good catch! I will review that again.

**Comment by samoht** [comment]
I've fixed the ns/second conversion bug.

**Pull request merged**